src/utils/form-data.js

```diff
@@ -43,3 +49,32 @@ export default async function * formDataIterator(form, boundary) {
 
 	yield `${dashes}${boundary}${dashes}${carriage.repeat(2)}`;
 }
+
+/**
+ * @param {FormData} form
+ * @param {string} boundary
+ */
+export function getFormDataLength(form, boundary) {
+	let length = 0;
+
+	for (const [name, value] of form) {
+		length += Buffer.byteLength(getHeader(boundary, name, value));
+
+		if (isBlob(value)) {
+			length = value.size;
+		} else {
+			length += Buffer.byteLength(value);
+		}
+
+		length += carriageLength;
+	}
+
+	// The FormData is empty?
+	if (length === 0) {
+		return length;
+	}
```

Comment:
Even if it is empty it should still append the boundary...
![Skärmavbild 2020-06-05 kl  01 15 14](https://user-images.githubusercontent.com/1148376/83819436-22763c00-a6ca-11ea-87a4-0be31ed93eed.png)

from native chrome...

Reply:
So what should I return here? A boundary's length?

Reply:
I took this code from here: https://github.com/octet-stream/form-data/blob/4e5c214fbbdeb3474cc5a3cc19a72e372f1ced1c/src/lib/FormData.mjs#L354-L356

Reply:
I'm pretty sure https://github.com/node-fetch/node-fetch/pull/603#issuecomment-638985881 calculates the size correct and dose it faster without the need to use some sort of getHeader function. guess it should be `--${boundary}--` size (4 + boundary.size)

Reply:
> I'm pretty sure #603 (comment) calculates the size correct and dose it faster without the need to use some sort of getHeader function. guess it should be --${boundary}-- size (4 + boundary.size)

I'm just using the same function for formDataIterator and getFormDataLength, so its in the name of DRY principle. Also, these literals looks meaningless when I'm reading your code. And I don't thinл is there some really necessary performance difference to avoid of getHeader.